### ISSUE TITLE
sourceComments option gets ignored if called by gulp.watch

### ISSUE THREAD

**Issue by hashworks** [dlmanning/gulp-sass#234]
Example:
```js
var gulp = require('gulp');
var sass = require('gulp-sass');

gulp.task('sass', function() {
	gulp.src('**/*.{scss,sass}')
		.pipe(sass({
			sourceComments: true,
			outputStyle: 'expanded',
			errLogToConsole: true
		}))
		.pipe(gulp.dest('screen.css'));
});

gulp.task('watch', ['sass'], function() {
	var sassWatcher = gulp.watch('**/*.{scss,sass}', ['sass']);
});
```

When I call `gulp sass` it generates source comments, as expected. But when I run `gulp watch` no source comments get generated.

**Comment by Snugug** [comment]
Please try with the latest version:

```
npm install gulp-sass@next --save-dev
```

**Comment by hashworks** [comment]
Same result.

**Comment by Snugug** [comment]
I cannot replicate your error using the latest version of `gulp-sass`, in fact. I very clearly see the comments, including on `watch`, but it brings up an interesting question we should resolve.

Because we're passing a string to `node-sass` and not a path, the `sourceComment` says it's coming from `stdin` for anything coming from the main file. Should we replace this with the path to that file before passing passing back along @dlmanning @Keats? `process.cwd()` replaced below for the absolute path to the Gulp file.

```css
/* line 1, {{process.cwd()}}/sass/_foo.scss */
.banana {
  content: 'waldo'; }

/* line 3, stdin */
.foo {
  content: 'bar'; }

/* line 7, stdin */
.bar {
  content: 'baz'; }

/* line 11, stdin */
.baz {
  content: 'baz'; }

/* line 15, stdin */
.qux {
  content: 'qux'; }
```

**Comment by Snugug** [comment]
I'm also happy to defer this to 2.1 as most people seem to be interested in Source Maps over Comments and I don't want to block the 2.0 release on this

**Comment by hashworks** [comment]
I'm ok with moving this to 2.1. I'll take a look at this again, I could bet in my case there where no source comments at all.

**Comment by Keats** [comment]
I didn't even know that was a thing (I use compressed output with no sourcemaps/anything).
We can definitely work on that for 2.1

**Comment by aphex** [comment]
For those of you, like me, who are to impatient to wait for a fix you can hack around this by tapping into the stream with [gulp-tap] (https://www.npmjs.com/package/gulp-tap) and just doing plain ol' string replacement. It will likely slow things down as we have to toString the buffer then convert it back but it gets you out of a bind. 

```javascript
gulp.src(config.files.scss)
.pipe(sass({
    sourceComments:true
}))
.pipe(tap(function(file, t) {
    file.contents = new Buffer(file.contents.toString().replace(/stdin \*\//g, file.path + ' */'));
}))
.pipe(concat('app.css'))
}))
```

**Comment by xzyfer** [comment]
This appears to be fix in v2.3.1.

**Comment by alehro** [comment]
In v2.3.2 it still doesn't work.
